internal/backend/sftp/foreground_solaris.go

```diff
@@ -0,0 +1,13 @@
+package sftp
+
+import (
+	"os/exec"
+
+	"github.com/restic/restic/internal/errors"
+)
+
+// golang on Solaris lacks SYS_IOCTL. It should be possible to workaround
+// that, however for turn these functions into no-ops.
+func startForeground(cmd *exec.Cmd) (bg func() error, err error) {
+	return nil, errors.New("sftp disabled on Solaris")
```

Comment:
It should be possible to use the sftp backend on solaris just fine, just without the magic that on Linux allows us to query for a password interactively and being able to tear down nicely at the same time.

Something like this should work:
```go
// run the command in its own process group
// so that SIGINT is not sent to it
cmd.SysProcAttr = &syscall.SysProcAttr{
    Setpgid: true,
}

// start the process
err = cmd.Start()
if err != nil {
    return nil, errors.Wrap(err, "cmd.Start")
}

bg = func() error { return nil }
return bg, nil
```